metaflow/plugins/kfp/kfp_cli.py

```diff
@@ -0,0 +1,165 @@
+    DEFAULT_KFP_YAML_OUTPUT_PATH,
+)
+from metaflow.plugins.kfp.kfp import KubeflowPipelines
+from metaflow.plugins.kfp.kfp_decorator import KfpInternalDecorator
+from metaflow.util import get_username
+
+
+class IncorrectMetadataServiceVersion(MetaflowException):
+    headline = "Incorrect version for metaflow service"
+
+
+@click.group()
+def cli():
+    pass
+
+
+@cli.group(name="kfp", help="Commands related to Kubeflow Pipelines.")
+@click.pass_obj
+def kubeflow_pipelines(obj):
+    obj.check(obj.graph, obj.flow, obj.environment, pylint=obj.pylint)
+
+
+@kubeflow_pipelines.command(help="Deploy a new version of this workflow to Kubeflow Pipelines.")
+@click.option(
+    "--experiment-name",
+    "experiment_name",
+    default=DEFAULT_EXPERIMENT_NAME,
+    help="the associated experiment name for the run",
+)
+@click.option(
+    "--run-name", "run_name", default=DEFAULT_RUN_NAME, help="name assigned to the new KFP run",
+)
+@click.option(
+    "--namespace", "namespace", default=KFP_SDK_NAMESPACE, help="namespace of your run in KFP.",
+)
+@click.option(
+    "--api-namespace",
+    "api_namespace",
+    default=KFP_SDK_API_NAMESPACE,
+    help="namespace where the API service is run.",
+)
+@click.option(
+    "--yaml-only",
+    "yaml_only",
+    is_flag=True,
+    default=False,
+    help="Generate the KFP YAML which is used to run the workflow on Kubeflow Pipelines.",
+)
+@click.option(
+    "--pipeline-path",
+    "pipeline_path",
+    default=DEFAULT_KFP_YAML_OUTPUT_PATH,
+    help="the output path (or filename) of the generated KFP pipeline yaml file",
+)
+@click.pass_obj
+def run(
+    obj,
+    experiment_name=DEFAULT_EXPERIMENT_NAME,
+    run_name=DEFAULT_RUN_NAME,
+    namespace=KFP_SDK_NAMESPACE,
+    api_namespace=KFP_SDK_API_NAMESPACE,
+    yaml_only=False,
+    pipeline_path=DEFAULT_KFP_YAML_OUTPUT_PATH,
+):
+    check_metadata_service_version(obj)
+    flow = make_flow(obj, current.flow_name, namespace, api_namespace)
+
+    if yaml_only:
+        pipeline_path = flow.create_kfp_pipeline_yaml(pipeline_path)
+        obj.echo(
+            "\nDone converting *{name}* to {path}".format(
+                name=current.flow_name, path=pipeline_path
+            )
+        )
+    else:
+        obj.echo("Deploying *%s* to Kubeflow Pipelines..." % current.flow_name, bold=True)
+        run_pipeline_result = flow.create_run_on_kfp(experiment_name, run_name)
+
+        obj.echo("\nRun created successfully!\n")
+        kfp_run_url = posixpath.join(
+            KFP_RUN_URL_PREFIX, "_/pipeline/#/runs/details", run_pipeline_result.run_id
+        )
+        obj.echo("{kfp_run_url}\n".format(kfp_run_url=kfp_run_url), fg="cyan")
+
+
+def check_metadata_service_version(obj):
+    metadata = obj.metadata
+    version = metadata.version()
+    if version == "local":
+        return
+    elif version is not None and LooseVersion(version) >= LooseVersion("2.0.2"):
+        # Metaflow metadata service needs to be at least at version 2.0.2
```

Comment:
Fixed by reusing the `step_functions_cli.py` version.
This is necessary because we are registering book-keeping metadata just as `step_functions_decorator.py` does:
https://github.com/zillow/metaflow/blob/a63e24e75f3e206d948a15158bff311c58767d1e/metaflow/plugins/kfp/kfp_decorator.py#L31-L38